addons/mail/static/src/new/core/message_reactions_model.js

```diff
@@ -0,0 +1,50 @@
+/** @odoo-module **/
+
+import { Partner } from "./partner_model";
+
+export class MessageReactions {
+    /** @type {string} */
+    content;
+    /** @type {number} **/
+    count;
+    /** @type {Partner} **/
+    partners = [];
+    /** @type {number} **/
+    messageId;
+
+    /**
+     * @param {import("@mail/new/core/messaging").Messaging['state']} state
+     * @param {Object} data
+     * @returns {MessageReactions}
+     */
+    static insert(state, data) {
+        let reaction =
+            state.messages[data.message.id] &&
+            state.messages[data.message.id].reactions.find(
+                ({ content }) => content === data.content
+            );
```

Comment:
```suggestion
        let reaction = state.messages[data.message.id]?.reactions.find(
            ({ content }) => content === data.content
        );
```

Reply:
Oh right, optional chaining, thanks for pointing that out :+1: